# Варіант 4

#### 1. З нормальної генеральної сукупності взято вибірку об’ємом $n=17$, знайдено її виправлену вибіркову дисперсію $s^2=0.24$. Перевірити при рівні значущості 0,05 нульову гіпотезу $H_0:\sigma^2=\sigma_0^2=0.18$, альтернативна гіпотеза $H_1:\sigma^2>0.18$

In [1]:
from sys import path
path.append('..')
import mathstat
n = 17
s12 = 0.24
sigma22 = 0.18
alpha = 0.05

Знайдемо спостережуване значення критерію $\chi_{спост.}=\dfrac{(n-1)s^2}{\sigma_0^2}$

In [2]:
chi = (n-1)*s12/sigma22
print(f"{chi=}")

chi=21.333333333333332


In [3]:
k = n - 1
print(f"Кількість ступенів вільності {k=}")

Кількість ступенів вільності k=16


За умовою, альтернативна гіпотеза має вигляд $H_1:\sigma^2>0.18$, тому критична область правостороння. По таблиці критичних точок розподілу $\chi^2$ за заданим рівнем значущості 0.05 і кількістю ступенів вільності k = 16 знаходтмо критичну точку. Замість таблиці використаємо модуль scipy.stats.chi2 оскільки в обох випадках дії по знаходженню - одникові

In [4]:
from scipy.stats import chi2
chi_critical = chi2.ppf(1-alpha, k)
print(f"Критична точка хі-квадрат = {chi_critical}")

Критична точка хі-квадрат = 26.29622760486423


In [5]:
print(chi<chi_critical)

True


$\chi_{спост.}<\chi_{крит.} 21.(3) < 26.3$ Отже немає підстав відхилити нульову гіпотезу

#### 2. Дві лабораторії однаковим методом в однаковій послідовності визначали вміст вуглецю в 13 зразках сталі. Отримані наступні результати аналізів:

| І |$x_i$| лабораторія | 0,18 | 0,12 | 0,12 | 0,08 | 0,08 | 0,12 | 0,19 | 0,32 | 0,27 | 0,22 | 0,34 | 0,14 | 0,46 |
|---|---|-------------|------|------|------|------|------|------|------|------|------|------|------|------|------|
|ІІ |$y_i$|лабораторія  | 0,16 | 0,09 | 0,08 | 0,05 | 0,13 | 0,10 | 0,14 | 0,30 | 0,31 | 0,24 | 0,28 | 0,11 | 0,42 |

#### На рівні значущості 0,05 з’ясувати, істотно чи неістотно відрізняються середні результати аналізів в припущенні, що вони розподілені нормально.


In [6]:
import numpy as np
n = 13
k = n-1
alpha = 0.05
x_i = np.array([0.18, 0.12, 0.12, 0.08, 0.08, 0.12, 0.19, 0.32, 0.27, 0.22, 0.34, 0.14, 0.46])
y_i = np.array([0.16, 0.09, 0.08, 0.05, 0.13, 0.10, 0.14, 0.30, 0.31, 0.24, 0.28, 0.11, 0.42])

Побудуємо допоміжну вибірку $d_i$

In [7]:
d_i = x_i-y_i
print("d_i =",d_i)

d_i = [ 0.02  0.03  0.04  0.03 -0.05  0.02  0.05  0.02 -0.04 -0.02  0.06  0.03
  0.04]


 Обчислимо вибіркове середнє $\bar{d}$ і виправлене середньоквадратичне відхилення $s_d$

In [8]:
d_sample_mean = mathstat.calc_sample_mean(d_i)
sd_corr_variance = mathstat.correct_variance(mathstat.calc_variance(d_i), n)
print(f"{d_sample_mean=} \n{sd_corr_variance=}")

d_sample_mean=0.017692307692307695 
sd_corr_variance=0.0011358974358974358


Обчислимо спостережуване значення критерію $T_{спост.}=\dfrac{\bar{d}\sqrt n}{s_d}$

In [9]:
T = d_sample_mean*mathstat.math.sqrt(n)/sd_corr_variance
print(f"{T=}")

T=56.15869932438269


За таблицею критичних точок розподілу Стьюдента, за рівнем значущості 0,05, вміщеним у верхньому рядку таблиці, і кількістю ступенів вільності k = 12 знаходимо критичну точку. Замість таблиці використаємо модуль scipy.stats.t але врахуємо, що цього разу критична область двобічна

In [10]:
from scipy.stats import t
T_critical = t.ppf(1-alpha/2, k)
print(f"Критична точка T({alpha},{k}) = {round(T_critical,2)}")

Критична точка T(0.05,12) = 2.18


$T_{спост.}>T_{критичне}\ \ 56.16>2.18$ Нульову гіпотезу відхиляєм